src/Crypto/KeyPair.php

```diff
@@ -0,0 +1,83 @@
+<?php declare(strict_types = 1);
+
+namespace Cmnty\Push\Crypto;
+
+use InvalidArgumentException;
+
+class KeyPair
+{
+    /**
+     * @var PrivateKey
+     */
+    private $private;
+
+    /**
+     * @var PublicKey
+     */
+    private $public;
+
+    /**
+     * Instantiate a new KeyPair from a private and public key.
+     *
+     * @param PrivateKey $private
+     * @param PublicKey $public
+     *
+     * @throws InvalidArgumentException When private and public key do not match.
+     */
+    public function __construct(PrivateKey $private, PublicKey $public = null)
```

Comment:
Can $public really be "null"? If so, update the docblock.

edit: Ah, reading further it indeed can be null.